server/conn.go

```diff
@@ -42,31 +42,9 @@ var baseConnID uint32 = 10000
 func NewConn(conn net.Conn, user string, password string, h Handler) (*Conn, error) {
 	p := NewInMemoryProvider()
 	p.AddUser(user, password)
+	server := NewDefaultServer()
```

Comment:
multiple `NewConn` will call multiple times of `NewDefaultServer`. We still need to cache the result of `NewConn`

Reply:
Maybe `NewConn` should be a method of `type Server`?